src/components/TextInput/TextInput.js

```diff
@@ -140,19 +149,30 @@ function PrivateTextInput({
           capitalize={capitalize}
         />
       )}
-      <input
-        type={type}
-        className={getClasses()}
-        disabled={disabled}
-        name={name}
-        placeholder={rest.placeholder}
-        onChange={onChange}
-        onBlur={onBlur}
-        value={value}
-        data-tid={rest['data-tid']}
-        aria-label={name} // https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute
-        autoComplete={autoComplete}
-      />
+      <div className={styles.TextInputWrapper}>
+        <input
+          type={type}
+          className={getClasses()}
+          disabled={disabled}
+          name={name}
+          placeholder={rest.placeholder}
+          onChange={onChange}
+          onBlur={onBlur}
+          value={value}
+          data-tid={rest['data-tid']}
+          aria-label={name} // https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute
+          autoComplete={autoComplete}
+        />
+        {((iconPrefix === 'fas' && iconName === 'lock') ||
+          (iconPrefix === 'far' && iconName === 'eye-slash')) && (
```

Comment:
Again, a user would have to check the source code here to see these are the only valid icon combinations, any reason we can't allow any valid prefix with any valid icon name?

Reply:
Resolving since we discussed on zoom.